### PR TITLE
Move cohttp_lwt_unix stuff to lwt_unix

### PR DESCRIPTION
**Pull request by rgrinberg**
One directory per findlib package seems like a good rule

### CODE REVIEW

File: _oasis [hunk 1]
```diff
@@ -94,7 +94,7 @@ Library cohttp_lwt
 Library cohttp_lwt_unix
   Build$:        flag(lwt_unix)
   Install$:      flag(lwt_unix)
-  Path:          lwt
+  Path:          lwt_unix
   Findlibname:   lwt
```
Comment (samoht) [inline _oasis]: I'm not sure to understand that change. The ocamlfind name is actually `lwt`.

Comment (rgrinberg) [inline _oasis]: I'm not changing the findlib structure of the library just moving the unix related stuff into its own directory.

**Comment by rgrinberg** [comment]
@samoht I took your advice and just split the current `lwt` dir to `lwt` and `lwt-core` to match the findlib subpackage. I think this is the least confusing although the naming isn't the best.

**Comment by avsm** [comment]
```
# [...]
#                               - Building lib/../lwt-core/Cohttp_lwt_body.mly
#                   - Building lib/../lwt-core/cohttp_lwt_body.odoc:
#                       - Failed to build all of these:
#                           - Building lib/../lwt-core/cohttp_lwt_body.ml:
#                               - Failed to build all of these:
#                                   - Building lib/../lwt-core/cohttp_lwt_body.mly
#                                   - Building lib/../lwt-core/cohttp_lwt_body.mll
#                           - Building lib/../lwt-core/cohttp_lwt_body.mli:
#                               - Building lib/../lwt-core/cohttp_lwt_body.mly
```

**Comment by rgrinberg** [comment]
Huh. Wondering why it works for me.

**Comment by rgrinberg** [comment]
Not sure how to fix this as I don't know enough ocamldoc to make it look in `lwt` and `lwt-core`.

**Comment by avsm** [comment]
This is not ideal, but just split out the docs.

```
diff --git a/_oasis b/_oasis
index d1333ba..ef7051b 100644
--- a/_oasis
+++ b/_oasis
@@ -146,6 +146,15 @@ Document cohttp
   XOCamlbuildPath:      lib
   XOCamlbuildLibraries: cohttp
 
+Document cohttp_lwt_core
+  Title:                Cohttp Lwt core docs
+  Type:                 ocamlbuild (0.4)
+  BuildTools+:          ocamldoc
+  Build$:               flag(lwt)
+  Install:              true
+  XOCamlbuildPath:      lwt-core
+  XOCamlbuildLibraries: cohttp.lwt-core
+
 Document cohttp_lwt
   Title:                Cohttp Lwt docs
   Type:                 ocamlbuild (0.4)
@@ -153,7 +162,7 @@ Document cohttp_lwt
   Build$:               flag(lwt_unix)
   Install:              true
   XOCamlbuildPath:      lwt
-  XOCamlbuildLibraries: cohttp.lwt-core, cohttp.lwt
+  XOCamlbuildLibraries: cohttp.lwt
 
 Document cohttp_async
   Title:                Cohttp Async docs
```

Once @dsheets is done with the codoc generation, we can get rid of ocamldoc entirely and have one cross-referenced doc set.

**Comment by rgrinberg** [comment]
Btw the docs are totally broken by camlp4 even without this PR unfortunately.

**Comment by avsm** [comment]
Yeah. It really is time to leave camlp4 behind. Waiting on @diml's type_conv ppx though.

**Comment by rgrinberg** [comment]
travis gives 1/3

good enough...

**Pull request merged**